transitions/core.py

```diff
@@ -359,16 +356,17 @@ def __init__(self, model=None, states=None, initial=None, transitions=None,
         self._transition_queue = deque()
         self.models = []
 
-        if initial is None:
-            self.add_states('initial')
+        if model is None and add_self:  # TODO(pbovbel) reconsider API for next major release
+            model = self
+
```

Comment:
I'd suggest to remove `add_self` again. Having accidentally set `add_self=False` and no model leads to weird effects which may be hard to track down. If no model is passed, use `self` as model. If someone wants to explicitly add self AND a model, why not do it in two steps `m = Machine(model); m.add_model(m)` . This keeps the consistency if previous versions and requires one line of code for a special case instead of requiring another parameter to consider for every other case. What do you think?